src/wallet/asyncrpcoperation_common.cpp

```diff
@@ -0,0 +1,58 @@
+#include "asyncrpcoperation_common.h"
+
+#include "core_io.h"
+#include "init.h"
+#include "rpc/protocol.h"
+#include "wallet.h"
+
+extern UniValue signrawtransaction(const UniValue& params, bool fHelp);
+
+UniValue SendTransaction(CTransaction& tx, bool testmode) {
+    UniValue o(UniValue::VOBJ);
+    // Send the transaction
+    if (!testmode) {
+        CWalletTx wtx(pwalletMain, tx);
+        pwalletMain->CommitTransaction(wtx, boost::none);
+        o.push_back(Pair("txid", tx.GetHash().ToString()));
+    } else {
+        // Test mode does not send the transaction to the network.
+        UniValue o(UniValue::VOBJ);
```

Comment:
This is a mistake - we should not be creating a new object here. This would have been caught when running the test cases, but I discovered this while making some of the @str4d's suggested changes. This will be fixed in a subsequent commit.